pkg/issuer/acme/acme.go

```diff
@@ -57,6 +57,9 @@ type Acme struct {
 
 	// metrics is used to create instrumented ACME clients
 	metrics *metrics.Metrics
+
+	// userAgent is the string used as the UserAgent when making HTTP calls.
+	userAgent string
```

Comment:
So the user agent used in calls to ACME and other external services will change from `cert-manager/<version>` ([here](https://github.com/jetstack/cert-manager/blob/master/pkg/util/useragent.go#L29)) to something like `cert-manager-<componentname>/<version> (<platform>) cert-manager/<gitCommit>`([here](https://github.com/JoshVanL/cert-manager/blob/controllers-user-agent/pkg/util/useragent.go#L34-L36)).
Do we think that external services need this much information, especially which component within cert-manager is making the call?
I guess in some cases this will cause there to be calls with _different_ user agents made from cert-manager to an external service (i.e from issuers controller as well as from a specific certificate requests controller). I guess this could help with debugging in certain cases, but in other cases might make it a little harder to monitor the total number of calls made from cert-manager.

Should we add this change to release note? I'm thinking about i.e https://github.com/jetstack/cert-manager/issues/4612 where they seemed to have been using user agent to monitor number of calls from cert-manager, so may want to know if user agent changes.

Reply:
> There have been bugs in the past where I do think having the component name in the request would have sped up debugging time significantly. #3777 is one such example.

That's a really good example 👍🏼 

Agree that it should be trivial for an external service to modify the parser given the user agent string format as it's introduced in this PR and the value of this seems to outweigh any possible breakage, so it seems to make sense to me as long as the change is communicated- which it is as you've written a really good release note in PR description

Reply:
Sorry for digging up ancient history, but I noticed that @jsha from Let's Encrypt asked us to not include the Git hash in the user-agent string:
 * https://github.com/cert-manager/cert-manager/issues/997
 * https://github.com/cert-manager/cert-manager/issues/407#issuecomment-432336806

> One more request: The User-Agent string for tagged releases looks like: go-acme/2 jetstack-cert-manager/v0.5.0-7924346bd84e41053cc508956b0a1b567c932416. Since that's a tagged release, the git commit is redundant and just consumes extra bytes in our logs. Could you omit that part of the User-Agent string for tagged releases?